pylint/checkers/refactoring.py

```diff
@@ -476,8 +478,118 @@ def visit_unaryop(self, node):
                              args=(node.as_string(), suggestion))
 
 
+def _is_len_call(node):
+    """Checks if node is len(SOMETHING)."""
+    return (isinstance(node, astroid.Call) and isinstance(node.func, astroid.Name) and
+            node.func.name == 'len')
+
+def _is_constant_zero(node):
+    return isinstance(node, astroid.Const) and node.value == 0
+
+def _node_is_test_condition(node):
+    # the node is an if, while, assert of if expression statement
+    return isinstance(node, (astroid.If, astroid.While, astroid.Assert, astroid.IfExp))
+
+
+class LenChecker(checkers.BaseChecker):
+    """Checks for incorrect usage of len() inside conditions. pep8 states:
+    For sequences, (strings, lists, tuples), use the fact that empty sequences are false.
+
+        Yes: if not seq:
+             if seq:
+
+        No: if len(seq):
+            if not len(seq):
+
+    Problems detected:
+    * if len(sequence):
+    * if not len(sequence):
+    * if len(sequence) == 0:
+    * if len(sequence) != 0:
+    * if len(sequence) > 0:
+    """
+
+    __implements__ = (interfaces.IAstroidChecker,)
+
+    # configuration section name
+    name = 'len'
+    msgs = {'W1801': ('Do not use `len(SEQUENCE)` as condition value',
```

Comment:
This should start with `C` as it's a coding convention (See https://pylint.readthedocs.io/en/latest/user_guide/output.html#source-code-analysis-section)